src/main/java/uk/nhs/digital/nhsconnect/lab/results/utils/ResourceFullUrlGenerator.java

```diff
@@ -0,0 +1,14 @@
+package uk.nhs.digital.nhsconnect.lab.results.utils;
+
+import lombok.NonNull;
+import org.hl7.fhir.dstu3.model.Resource;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ResourceFullUrlGenerator {
+    private static final String FULL_URL_PREFIX = "urn:uuid:";
+
+    public String generateFullUrl(@NonNull final Resource resource) {
```

Comment:
Can this be simplified since we know we're generating a full url?
```suggestion
    public String generate(@NonNull final Resource resource) {
```

Reply:
Also is it really necessary to pass in the full resource when we only use the id?

Reply:
Not "necessary", but I feel "appropriate". It's generating the full url for a resource, so we give it the resource. The fact it does it with the result of `resource.getId()` is just how it's implemented.

What do you think?

Reply:
Ok, that's fair enough, happy to keep it as is 👍